include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -139,15 +139,13 @@ class pairwise_alignment_algorithm : protected policies_t...
     //!\endcond
     void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
     {
-        using result_value_t = typename alignment_result_value_type_accessor<alignment_result_type>::type;
         using std::get;
 
-        for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
+        for (auto && indexed_sequence_pair : indexed_sequence_pairs)
         {
-            result_value_t res{};
-            res.id = idx;
-            res.score = compute_matrix(get<0>(sequence_pair), get<1>(sequence_pair));
-            callback(alignment_result_type{res});
+            auto && sequence_pair = get<0>(indexed_sequence_pair);
```

Comment:
```suggestion
            auto & sequence_pair = get<0>(indexed_sequence_pair);
```

Isn't this guaranteed? Because we call it with lvalue `indexed_sequence_pair`?

Reply:
Not exactly: If std::tuple_element_t<0, pair_t> == something && then it would be binding a lvalue reference (after applying reference collapsing rules) to something that was an rvalue reference. Would that make sense?

Reply:
The problem I have with `&&` is that we might move data out, the following call gets the tuple again, so that means me pass a tuple that might have moved out data.

Reply:
How do we solve this? The sequence builder will need this data inside the `make_result_...`, or?

Reply:
That is a really good point. The only solution would be to pass the sequence pair and the id separatey into the call. I wanted to keep the parameter list small but maybe this is better in this case.

Reply:
Still open